convertMQTTTopicToRegex.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * This is the shared prefix name specified by the MQTT spec. When you
+ * subscribe to a topic, if you follow the format $share/group_name/topic_name,
+ * the messages will be dispersed to ONE consumer subscribed to that topic_name,
+ * and the incoming topic name will be "topic_name".
+ * @type {string}
+ */
+const sharedPrefix = '$share/';
+
+function convertMQTTTopicToRegex(topic) {
+    let strippedTopic = topic;
+
+    if (strippedTopic.startsWith(sharedPrefix)) {
+        let mangledTopic = strippedTopic.slice(sharedPrefix.length);
+        // do we have a group? It'll be shown here.
+        const indexOfTopic = mangledTopic.indexOf('/');
+
+        if (indexOfTopic > 0) {
```

Comment:
It is not clear why we do this replacement conditionally. We test only for cases when we have at least one '/' and it should be not the first one. Shouldn't we always do this replacement?

Like

```
const sharedPrefixRe = /^\$share\/[^/]*\//; 
const strippedTopic = topic.replace(sharedPrefixRe, '')
```

Reply:
Again, I'd have to test. But don't forget the incoming topic WILL NOT have $share prepended.